server/utils/parse_helper.py

```diff
@@ -110,6 +125,47 @@ def reparse_cleanup(project_details, user_id):
     delete_folder(directory)
 
 
+def extract_repository_metadata(repo):
+    languages = repo.get_languages()
+
+    total_bytes = sum(languages.values())
+    latest_commit = repo.get_commits()[0]
+
+    metadata = {
+        "basic_info": {
+            "full_name": repo.full_name,
+            "description": repo.description,
+            "created_at": repo.created_at.isoformat(),
+            "updated_at": repo.updated_at.isoformat(),
+            "default_branch": repo.default_branch,
+        },
+        "metrics": {
+            "size": repo.size,
+            "stars": repo.stargazers_count,
+            "forks": repo.forks_count,
+            "watchers": repo.watchers_count,
+            "open_issues": repo.open_issues_count,
+            "open_pull_requests": len(list(repo.get_pulls(state='open'))),
+        },
+        "languages": {
+            "breakdown": languages,
+            "total_bytes": total_bytes,
+        },
+        "commit_info": {
+            "total_commits": repo.get_commits().totalCount,
+            "last_commit_date": latest_commit.commit.author.date.isoformat(),
+            "last_commit_sha": latest_commit.sha,
+        },
```

Comment:
are we updating this at every commit?

Reply:
No, but I'll remove this since we're storing the latest commit in the table anyway.